packages/compass-connections/src/stores/active-connections.ts

```diff
@@ -0,0 +1,40 @@
+import type { ConnectionInfo } from '@mongodb-js/connection-info';
+import { useCallback, useEffect, useState } from 'react';
+import { useConnectionRepositoryContext } from '@mongodb-js/connection-storage/provider';
+import {
+  ConnectionsManagerEvents,
+  useConnectionsManagerContext,
+} from '../provider';
+
+export function useActiveConnections(): ConnectionInfo[] {
+  const connectionManager = useConnectionsManagerContext();
+  const connectionRepository = useConnectionRepositoryContext();
+
+  const [activeConnections, setActiveConnections] = useState<ConnectionInfo[]>(
+    []
+  );
+
+  const updateList = useCallback(async () => {
+    const list = [
+      ...(await connectionRepository.listFavoriteConnections()),
+      ...(await connectionRepository.listNonFavoriteConnections()),
+    ].filter(({ id }) => connectionManager.statusOf(id) === 'connected');
+    setActiveConnections(list);
+  }, []);
```

Comment:
useConnections as it is is already doing a lot of things and this would make it even more complex, our approach is to have smaller hooks, @kmruiz is doing something similar with the `useConnectionStatus`.
 the point about the list updating more often than necessary is valid though. perhaps I could include a check that compares the arrays before updating the state?

Reply:
Right, I forgot that useConnections is not actually a hook that gives you connections, but all the connections, local UI state, and the connect logic, you're correct

Reply:
To your second point: yes, We'd need to do some sort of deep compare here to avoid unnecessary re-renderes. The way it is currently implmented where connections are re-read from disk every time they are updated we can't really rely on referential equality a-la redux for those kinds of checks.

Reply:
Another option is to render the child components always, but display them only when they are active. It would work essentially as a virtualised list and we would isolate rendering reasons.

1. When you change the list of connections, it will just change the amount of list elements. If the keys are the same, this is really cheap.
2. When you change the status of the connection it will only rerender the list element for that connection.

You could do that by just using useConnectionStatus in the child component and using useConnections in the parent (or having a hook that does that independently but is reactive to changes on the storage like this https://github.com/mongodb-js/compass/pull/5584/files#diff-5f0d1b5898d50c4399d08b4b53a9f4d06a8e65aacdc585183dc78fa3cd418b65R301

Reply:
As always with delegating rendering logic to the child components themselves, you are essetially removing the ability to rely on React component lifecycles, this always requires more work to work consistently in the long run. Especially in this case where the rendering will be delegated eventually to another component, this would definitely lead to conflicts in which component owns the rendering. I would not pursue it.

Reply:
I'll go with the deep comparison, the ConnectionInfo isn't that complex and this would only happen if the connections or connection status changes. Apart of the lifecycle point, the object references not being the same might still cause unnecessary re-renders down the line.

Reply:
Connection info options can contain bson values, so I'd double-check that those are not tripping the comparison. There are some helpers in the bson packages to make the deep eq check IIRC